docs/fsharp/language-reference/plaintext-formatting.md

```diff
@@ -0,0 +1,434 @@
+be used to specify surrogate text and property to display a value. For example:
+
+```fsharp
+[<StructuredFormatDisplay("Counts({Clicks})")>]
+type Counts = { Clicks:int list}
+
+printfn "%20A" {Clicks=[0..20]}
+```
+
+produces
+
+```console
+Counts([0; 1; 2; 3;
+        4; 5; 6; 7;
+        8; 9; 10; 11;
+        12; 13; 14;
+        15; 16; 17;
+        18; 19; 20])
+```
+
+### Customize plain text formatting by overriding `ToString`
+
+The default implementation of `x.ToString()` is observable in F# programming. Often, the default results
+aren't suitable for use in either programmer-facing information display or user output, and as a result it
+is common to override the default implementation.  
+
+By default, F# record and union types override the implementation of `x.ToString()` with an implementation that
+uses `sprintf "%+A"`.  For example,
+
+```fsharp
+type Counts = { Clicks:int list }
+
+printfn "%s" ({Clicks=[0..10]}.ToString())
+```
+
+produces
+
+```console
+{ Clicks = [0; 1; 2; 3; 4; 5; 6; 7; 8; 9; 10] }
+```
+
+For class types, no default implementation of `ToString()` is provided and the .NET default is used, which reports the
+name of the type. For example,
+
+```fsharp
+type MyClassType(clicks: int list) =
+   member x.Clicks = clicks
+
+let data = [ MyClassType([1..5]); MyClassType([1..5]) ]
+printfn "Default structured print gives this:\n%A" data
+printfn "Default ToString gives:\n%s" (data.ToString())
+```
+
+produces
+
+```console
+Default structured print gives this:
+[MyClassType; MyClassType]
+Default ToString gives:
+[MyClassType; MyClassType]
+```
+
+Adding an override for `ToString()` can give better formatting.
+
+```fsharp
+type MyClassType(clicks: int list) =
+   member x.Clicks = clicks
+   override x.ToString() = sprintf "MyClassType(%0A)" clicks
+
+let data = [ MyClassType([1..5]); MyClassType([1..5]) ]
+printfn "Now structured print gives this:\n%A" data
+printfn "Now ToString gives:\n%s" (data.ToString())
+```
+
+produces
+
+```console
+Now structured print gives this:
+[MyClassType([1; 2; 3; 4; 5]); MyClassType([1; 2; 3; 4; 5])]
+Now ToString gives:
+[MyClassType([1; 2; 3; 4; 5]); MyClassType([1; 2; 3; 4; 5])]
+```
+
+## F# Interactive structured printing
+
+F# Interactive (`dotnet fsi`) uses an extended version of structured plain text formatting to
+report values and allows additional customizability. For more information, see [F# Interactive](fsharp-interactive-options.md).
+
+## Customize debug displays
+
+Debuggers for .NET respect the use of attributes such as `DebuggerDisplayAttribute` and `DebuggerTypeProxyAttribute`,
+and these affect the structured display of objects in debugger inspection windows.
+The F# compiler automatically generated these attributes for discriminated union and record types, but
+not class, interface, or struct types.
+
+These attributes are ignored in F# plain text formatting, but it can be useful to implement
+these methods to improve displays when debugging F# types.
```

Comment:
Is it worth giving an example here? It's quite the parlor trick, e.g. https://github.com/dotnet/fsharp/blob/master/src/fsharp/FSharp.Core/prim-types.fs#L3528-L3575

Reply:
I think we should write a separate article on `Making your F# code easier to debug`. It's not specifically to do with plain text formatting - though important enough to call out that plain text formatting specifically doesn't use these attributes,though the default implementation of these attributes *does* use plain text formatting 

It's hard to give an example without showing screenshots I think and that would be inappropriate here,